Converting to Swift 3

plivesey opened the pull request:
This keeps all the APIs the same as before and only updates the project.
In a future commit, we'll convert the APIs.

Commit:
Initial conversion to swift 3
Unit tests pass

Commit:
Converting the Sample App

Commit:
Simplifying an enum case call

plivesey commented:
This needs to be done after we do the Swift 2.3 merge.

nicksnyder commented on RocketData/BatchDataProviderListener.swift:
```diff
@@ -24,23 +24,23 @@ import ConsistencyManager
 
     /// This is the batch listener from the consistency manager which contains most of the logic for doing batch listening
-    private let batchListener: BatchListener
+    fileprivate let batchListener: BatchListener
```
I know it is a pain, but I would audit all fileprivates to see if they can be replaced with private. Probably easiest to do a global search and replace and then add back any fileprivates that are necessary.

plivesey replied:
Not a pain. Pretty easy.
Sort of a funny task though because I'll just change them back if I need to.

plivesey replied:
But agree, may as well do this.

nicksnyder commented on SampleApp/Podfile:
```diff
@@ -5,3 +5,14 @@ target :SampleApp do
   pod 'PINCache', :git => 'https://github.com/pinterest/PINCache.git', :commit => '58635e4ff8d00fcd25084259a625c0615e9d071a'
 end
 
+# This is necessary to convert the target to swift 2.3
```
3.0

Commit:
Changing as much to private as I can
Docs fix in Podfile

Commit:
Merge branch 'master' of https://github.com/linkedin/RocketData into swift3

Commit:
Upgrading to xcode 8

Commit:
Changing the SDK and adding tests for iOS 10
Likely it will fail because the iOS 8 simulators are flaky, but going to try it

Commit:
Removing an errant space in build.sh causing failures

Commit:
Cleaning up the Podfile after a bad merge

Commit:
Disabling iOS 8.4 because it isn't very stable

plivesey commented:
Going to merge because on of the checks passed. Looks like the other is travis problems.
